File: src/main/java/io/pageobject/generator/name/ButtonPostfixNamePostProcessor.java
package io.pageobject.generator.name;

import org.jsoup.nodes.Element;

import static io.pageobject.generator.element.Elements.isButton;

public class ButtonPostfixNamePostProcessor extends PostfixNamePostProcessor {

    public ButtonPostfixNamePostProcessor() {
        super("Button", new String[] {"Button"});
    }

    @Override
    public boolean isMatchingElement(Element element) {
        return isButton(element);
    }
}

File: lib/name/button-postfix-name-post-processor.js
"use strict";

const PostfixNamePostProcessor = require('./postfix-name-post-processor');
const isButton = require('../dom/elements').isButton;

class ButtonPostfixNamePostProcessor extends PostfixNamePostProcessor {

  constructor() {
    super('Button', ['Button']);
  }

  isMatchingElement(element) {
    return isButton(element);
  }

}

module.exports = ButtonPostfixNamePostProcessor;